### PR TITLE
Docs: Add readme for networking tests

### LINKED ISSUE #931: Add README for networking tests
The main intention of this readme is to describe the current tests that we have (which tools are using and why the are important) as well as how to run them.

### PR DESCRIPTION
**Pull request by GabyCT**
Add readme for tests/metrics/networking.

Fixes #931.

Signed-off-by: Gabriela Cervantes <gabriela.cervantes.tellez@intel.com>

### CODE REVIEW

**Comment by chavafg** [comment]
qa-passed

**Comment by chavafg** [comment]
qa-passed

File: tests/metrics/network/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+Measures bidirectional network bandwidth using iperf. Bidirectional tests are used when 
+we want to test both servers for the maximum amount of throughput.
+
+- network-latency
+
+Measures latency using ping. The ping utility can be used as a tool to measure how long 
+it takes one packet to get from one point to another.
+
+- network-metrics-iperf3
```
Comment (jodh-intel) [inline tests/metrics/network/README.md]: Could you explain how this test differs from the `network-metrics` test which also measures bandwidth using `iperf` (not `iperf3`)?

File: tests/metrics/network/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+with iperf. PSS is a meaningful representation of memory usage by applications.
+
+- network-metrics-memory-rss-1g
+
+Measures Resident Set Size (RSS) memory using smem while running a transfer rate of 1 Gb with nuttcp.
+RSS is a standard measure to monitor memory usage in a physical memory scheme.
+
+## Running the networking tests
+
+Execute `sudo -E PATH=$PATH make metrics tests` to generate the necessary files to run individual tests.
```
Comment (jodh-intel) [inline tests/metrics/network/README.md]: Should this be `make metrics-tests`?

File: tests/metrics/network/README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+```
+$ cd tests/metrics
+$ bash network/network-metrics.sh
```
Comment (jodh-intel) [inline tests/metrics/network/README.md]: Does this need to be run using `sudo`?

File: tests/metrics/network/README.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+- Bandwidth
+- Latency
+- Jitter
+- Throughput
+- CPU % and memory consumption
+
+## Performance tools
+
+- Iperf/Iperf3 is a simple tool to let you measure the bandwidth and the quality of a network
```
Comment (jodh-intel) [inline tests/metrics/network/README.md]: What is the difference between `iperf` and `iperf3`?

Comment (GabyCT) [inline tests/metrics/network/README.md]: @jodh-intel, I put a comment about the difference

File: tests/metrics/network/README.md (hunk 1 above)
Comment (GabyCT) [inline tests/metrics/network/README.md]: @jodh-intel , I also put here a comment thanks

**Comment by jodh-intel** [comment]
Hi @rcaballeromx - could you take a look at this please?

lgtm

**Comment by chavafg** [comment]
qa-passed

**Comment by chavafg** [comment]
qa-passed

File: tests/metrics/network/README.md (hunk 1 above)
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: My feedback regarding network-metrics applies to all tests.

File: tests/metrics/network/README.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+used when we want to test both servers for the maximum amount of throughput.
+
+- network-latency
+
+Measures latency using ping. The ping utility can be used as a tool to measure how long 
+it takes one packet to get from one point to another.
+
+- network-metrics-iperf3
+
+Measures bandwidth, jitter and bidirectional bandwidth using iperf3 (single threaded). The
```
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: Change to:
`network-metrics-iperf3` measures bandwidth, jitter, and bidirectional bandwidth using iperf3 on single threaded connections.

File: tests/metrics/network/README.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+- network-latency
+
+Measures latency using ping. The ping utility can be used as a tool to measure how long 
+it takes one packet to get from one point to another.
+
+- network-metrics-iperf3
+
+Measures bandwidth, jitter and bidirectional bandwidth using iperf3 (single threaded). The 
+bandwidth test will help to show the rate of speed with which the data is transferred. On the other hand,
```
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: The bandwidth tests show the speed of the data transfer.

File: tests/metrics/network/README.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+- network-latency
+
+Measures latency using ping. The ping utility can be used as a tool to measure how long 
+it takes one packet to get from one point to another.
+
+- network-metrics-iperf3
+
+Measures bandwidth, jitter and bidirectional bandwidth using iperf3 (single threaded). The 
+bandwidth test will help to show the rate of speed with which the data is transferred. On the other hand, 
+the jitter test will measure the variation in the delay of received packets.
```
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: On the other hand, the jitter test measures the variation in the delay of received packets.

File: tests/metrics/network/README.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+- network-metrics-nuttcp
+
+Measures UDP bandwidth using nuttcp. This will help to show the rate of speed which the data
```
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: `network-metrics-nuttcp` measures the UDP bandwidth using nuttcp. This tool shows the speed of the data transfer for the UDP protocol.

File: tests/metrics/network/README.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+is transferred for the UDP protocol.
+
+- network-nginx-ab-benchmark
+
+Measures network performance using an nginx container and running the Apache benchmarking
```
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: `network-nginx-ab-benchmark` measures the network performance. It uses an nginx container and runs the Apache benchmarking tool in the host to calculate the requests per second.

File: tests/metrics/network/README.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+tool in the host to calculate the requests per second.
+
+- network-metrics-cpu-consumption
+
+Measures CPU % consumption while running maximum network bandwidth with iperf.
```
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: `network-metrics-cpu-consumption` measures the percentage of CPU consumption used while using the maximum network bandwidth with iperf.

File: tests/metrics/network/README.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+- network-metrics-memory-pss
+
+Measures Proportional Set Size (PSS) memory using smem while running maximum network bandwidth
```
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: `network-metrics-memory-pss` measures the Proportional Set Size (PSS) memory. It uses smem while running the maximum network bandwidth with iperf. PSS is a meaningful representation of the memory applications use.

File: tests/metrics/network/README.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+with iperf. PSS is a meaningful representation of memory usage by applications.
+
+- network-metrics-memory-rss-1g
+
+Measures Resident Set Size (RSS) memory using smem while running a transfer rate of 1 Gb with nuttcp.
```
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: `network-metrics-memory-rss-1g` measures the Resident Set Size (RSS) memory using smem while running a transfer of one Gb with nuttcp. RSS is a standard measure to monitor the memory usage in a physical memory scheme.

File: tests/metrics/network/README.md [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+RSS is a standard measure to monitor memory usage in a physical memory scheme.
+
+## Running the networking tests
+
+Execute `sudo -E make metrics tests` to generate the necessary files to run individual tests. In order
```
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: 1. Generate the necessary files to run the individual tests with the following command:
\`\`\`
sudo -E make metrics tests
\`\`\`

File: tests/metrics/network/README.md [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+
+- network-metrics-memory-rss-1g
+
+Measures Resident Set Size (RSS) memory using smem while running a transfer rate of 1 Gb with nuttcp.
+RSS is a standard measure to monitor memory usage in a physical memory scheme.
+
+## Running the networking tests
+
+Execute `sudo -E make metrics tests` to generate the necessary files to run individual tests. In order 
+to run the networking tests please use `sudo`.
```
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: 2. Run the networking tests using sudo with the following commands:

**Comment by rcaballeromx** [comment]
@jodh-intel I just submitted my review.

**Comment by GabyCT** [comment]
@rcaballeromx thanks for the feedback, changes were applied

**Comment by chavafg** [comment]
qa-passed

File: tests/metrics/network/README.md [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+## Networking tests
+
+### `network-metrics`
+
+`network-metrics` measures the bidirectional network bandwidth using iperf for multi-threaded connections. 
+Bidirectional tests are used to test both servers for the maximum amount of throughput.
+
+### `network-latency`
+
+`network-latency` measures the latency using ping. The ping utility measureshow long
```
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: measures how

File: tests/metrics/network/README.md [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+
+## Performance tools
+
+- Iperf and Iperf3 are simple tools to measure the bandwidth and the quality of a network 
+link. Iperf is multi-threaded and Iperf3 is single threaded.
+
+- Nuttcp is a network performance tool used to determine the raw UDP network layer 
+throughput.
+
+- Smem provides users with  diverse reports on memory usage.
```
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: with diverse
Double space

File: tests/metrics/network/README.md [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+
+### `network-latency`
+
+`network-latency` measures the latency using ping. The ping utility measureshow long 
+it takes one packet to get from one point to another.
+
+### `network-metrics-iperf3`
+
+`network-metrics-iperf3` measures bandwidth, jitter and bidirectional bandwidth using iperf3 on single threaded connections. The 
+bandwidth test show the speed of the data transfer. On the other hand,
```
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: shows

File: tests/metrics/network/README.md [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+transfer for the UDP protocol.
+
+### `network-nginx-ab-benchmark`
+
+`network-nginx-ab-benchmark` measures the network performance. It uses an nginx container and runs the Apache benchmarking
+tool in the host to calculate the requests per second.
+
+### `network-metrics-cpu-consumption`
+
+`network-metrics-cpu-consumption` measures the percentage  of CPU consumption used while the maximum network bandwidth with iperf.
```
Comment (rcaballeromx) [inline tests/metrics/network/README.md]: percentage of
Double space.

**Comment by GabyCT** [comment]
@rcaballeromx thanks, changes applied

**Comment by chavafg** [comment]
qa-passed

**Pull request merged**